src/crafting.cpp

```diff
@@ -1094,7 +1086,7 @@ void player::complete_craft( item &craft, const tripoint &loc )
         // Points to newit unless newit is a non-empty container, then it points to newit's contents.
         // Necessary for things like canning soup; sometimes we want to operate on the soup, not the can.
         item &food_contained = ( newit.is_container() && !newit.contents.empty() ) ?
-                               newit.contents.back() : newit;
+                               newit.contents.only_item() : newit;
```

Comment:
note for later: this is suspicious

Reply:
well, crafting pretty much only creates only one item in a container, at least for the forseeable future. worst case we get a debug message when something weird happens using only_item()